src/action.js

```diff
@@ -6,59 +6,62 @@ const registerDependencyAndroid = require('./android/registerNativeModule');
 const registerDependencyIOS = require('./ios/registerNativeModule');
 const copyAssetsAndroid = require('./android/copyAssets');
 const copyAssetsIOS = require('./ios/copyAssets');
+const pjson = require(path.join(process.cwd(), './package.json'));
 
+/**
+ * Namespace log messages
+ */
 log.heading = 'rnpm-link';
 
 /**
  * Returns an array of dependencies that should be linked/checked.
+ * @param {Object} pconfig Project config
  */
-const getProjectDependencies = () => {
-  const pjson = require(path.join(process.cwd(), './package.json'));
-  return Object.keys(pjson.dependencies).filter(name => name !== 'react-native');
-};
+const getProjectDependencies = (pconfig) =>
+  Object.keys(pconfig.dependencies).filter(name => name !== 'react-native');
 
 /**
  * Updates project and linkes all dependencies to it
- *
- * If optional argument [packageName] is provided, it's the only one that's checked
+ * @param {Object} config Composed configuration object
+ * @param {Array}  args   Arguments from the command line
  */
 module.exports = function link(config, args) {
-  const project = config.getProjectConfig();
+  const pconfig = config.getProjectConfig();
 
-  if (!project) {
+  if (!pconfig) {
     log.error('ERRPACKAGEJSON', `No package found. Are you sure it's a React Native project?`);
     return;
   }
 
   const packageName = args[0];
-  const dependencies = packageName ? [packageName] : getProjectDependencies();
+  const dependencies = packageName ? [packageName] : getProjectDependencies(pjson);
 
   dependencies
     .forEach(name => {
-      const dependencyConfig = config.getDependencyConfig(name);
```

Comment:
I am not actually sure if the naming changes are improving readability. I would've left the `project` and rename `dependencyConfig` to `dependency`.